back/engines/commercial/insights/spec/models/insights/category_spec.rb

```diff
@@ -32,12 +32,23 @@
   end
 
   describe 'associations' do
-    context 'when its view is deleted' do
-      subject(:category) { create(:category) }
+    subject(:category) { create(:category) }
 
+    context 'when its view is deleted' do
       before { category.view.destroy! }
 
       it { expect { category.reload }.to raise_error(ActiveRecord::RecordNotFound) }
     end
+
+    context 'when it is deleted' do
+      it 'gets deleted from zeroshot-classification tasks' do
+        task = create(:zsc_task, categories: [category])
+
+        aggregate_failures 'checking task' do
+          expect { category.destroy! }.to change { task.categories.count }.by(-1)
```

Comment:
We might want to delete the task (and ultimately request nlp to cancel it) if there's no categories left, which might happen because of the reset behavior (conceptually would be good for inputs too, but shouldn't happend as often as it might with categories).

Reply:
Ah yeah, I didn't think about tasks cancellation. Good point 🙂 I propose to log this as an improvement.